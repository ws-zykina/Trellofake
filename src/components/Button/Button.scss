.main-button {
  height: 30px;
  background-color: hsla(0, 0%, 100%, 0.3);
  min-width: 60px;
  color: white;
  font-size: 14px;
  border: none;
  border-radius: 4px;
}
.circle-button {
  height: 34px;
  width: 34px;
  border-radius: 50%;
  padding: 0;
  overflow: hidden;
  border: none;
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 14px;
  font-weight: bold;
}
.login-button {
}
.delete-button {
  
  height: 20px;
  width: 20px;
  display: inline-flex;
  border-radius: 5px;
  justify-content: center;
  align-items: right;
  background-color: rgb(80, 74, 74);
}
