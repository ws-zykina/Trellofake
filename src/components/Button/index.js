import React from 'react'
import PropTypes from 'prop-types';
import './Button.scss'

const Button = ({onClick, title, children, className}) => {
  return (
	<button onClick={onClick} className={className ? className : 'main-button'}>
		{children}
	</button>
  )
}


Button.propTypes = {
	title: PropTypes.string,
	onClick:  PropTypes.func.isRequired
};

Button.defaultProps = {
	children: 'Button'
};
export default Button;