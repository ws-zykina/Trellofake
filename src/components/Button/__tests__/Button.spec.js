import React from 'react';
import { shallow, mount } from 'enzyme';
import Button from '../';
describe('Button component', () => { 

    it('Should be styled correctly', () => {
        const component = shallow(<Button onClick={()=>{}}>123</Button>);
        expect(component).toMatchSnapshot();
    });

    it('Should be named correctly', () => {
        const component = shallow(<Button onClick={()=>{}}>123</Button>);
        expect(component.text()).toEqual('123')
    });

    it('Should have onclick', () => {
        const mockCallBack = jest.fn();

        const component = shallow(<Button onClick={mockCallBack}>123</Button>);
        component.find('button').simulate('click');
        expect(mockCallBack.mock.calls.length).toEqual(1);
    });

    it('Should have default title', () => {
        const component = shallow(<Button onClick={()=>{}} />);
        console.log(component.debug())
        expect(component.text()).toEqual('Button')
    });

})