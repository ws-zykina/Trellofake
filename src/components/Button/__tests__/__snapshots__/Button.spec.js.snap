// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Should be styled correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    onClick={[Function]}
  >
    123
  </Button>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "123",
      "className": "main-button",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": "123",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "123",
        "className": "main-button",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": "123",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
